### DISCUSSION TITLE
Has anyone got undercurl working on Windows with WSL?

### DISCUSSION THREAD

**Discussion by markhagemann** [wezterm/wezterm#981] [Q&A] [answered]
When I tested my dotfiles on Garuda OS in a VirtualBox using Tilix, the undercurl displayed as expected.

I've gone through a few guides without any success when running it though the Windows version of WezTerm though. I am guessing it has something to do with `conhost.exe` not supporting it yet?

**Comment by wez** [comment, marked as answer]
closing the loop; this is pending support in the Windows terminal guts and is tracked here:
https://github.com/microsoft/terminal/issues/7228
Once support lands in that repo, we can rebuild our bundled conpyt.dll and openconsole.exe to take advantage of it.

**Comment by wez** [comment]
Note that if you use eg: `wezterm ssh` to connect to a remote host, undercurl will work because conpty/openconsole are not involved in that case.

**Reply by unphased** [reply]
I'd like to understand this. I was having trouble with underline colors, and was surprised to find that it was not windows' SSH causing the issue (btw I'm using a newer version from scoop, so as to get working ctrl+space (null char) functionality over ssh. This worked great under Windows Terminal until I wanted undercurls for nvim). 

indeed wezterm ssh works. So it leads to the question: What does conpty do and why does wezterm use it if there is a way to run it without using it? Maybe another form of the same question is, how to avoid having to go over ssh if we just want to connect wezterm to our local WSL...

**Reply by wez** [reply]
https://wezfurlong.org/wezterm/what-is-a-terminal.html#windows-and-conpty has more information.

There are two ways to bypass conpty with wezterm:
* Use `wezterm ssh`
* Use [multiplexing](https://wezfurlong.org/wezterm/multiplexing.html#connecting-into-windows-subsystem-for-linux)

**Reply by unphased** [reply]
Thank you, makes sense -- i remember now that opening it up by default sees it run cmd.exe.

**Comment by ovidiu** [comment]
>  Once support lands in that repo, we can rebuild our bundled conpyt.dll and openconsole.exe to take advantage of it.

https://github.com/microsoft/terminal/issues/7228 is closed as fixed. Does this mean we may get this in one of the next versions?

**Reply by ovidiu** [reply]
Nevermind, I found my answer in [another discussion (4989)](https://github.com/wez/wezterm/discussions/4989). I switched to nightly and the undercurls work perfectly. 🎉
